### PR TITLE
feat(E2E): add end to end tests for the friends.

### PR DESCRIPTION
**Pull request by AdrienRenggli**
This PR adds an End-To-End test on the friend feature.

Furthermore it slightly modifies `MainActivity` and `ProfileSettingsScreen` to fix some minor issues and improve resistence to edge cases

### CODE REVIEW

File: app/src/androidTest/java/com/github/swent/swisstravel/e2e/E2EFriendsTest.kt [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+import com.github.swent.swisstravel.utils.FirestoreSwissTravelTest
+import com.google.firebase.Timestamp
+import kotlinx.coroutines.runBlocking
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+/**
+ * Done with the help of AI.
```
Comment (JstnFv) [inline app/src/androidTest/java/com/github/swent/swisstravel/e2e/E2EFriendsTest.kt]: Nice structure on the test, I think this is a good proof of an end-to-end test. Adding it to the documentation makes it clear too.

File: app/src/main/java/com/github/swent/swisstravel/MainActivity.kt [hunk 2]
```diff
@@ -130,6 +133,23 @@ class MainActivity : ComponentActivity() {
...
+  val auth = remember { FirebaseAuth.getInstance() }
+  val currentUser = remember { mutableStateOf(auth.currentUser) }
+
+  DisposableEffect(auth) {
+    val listener =
+        FirebaseAuth.AuthStateListener { authState -> currentUser.value = authState.currentUser }
+    auth.addAuthStateListener(listener)
+    onDispose { auth.removeAuthStateListener(listener) }
+  }
+  Log.d("rememberCurrentUser", "Current user: ${currentUser.value}")
```
Comment (JstnFv) [inline app/src/main/java/com/github/swent/swisstravel/MainActivity.kt]: ## Change
I don't know if we should keep the logs during the tests, in doubt I would suggest getting rid of this line.

File: app/src/androidTest/java/com/github/swent/swisstravel/e2e/E2EFriendsTest.kt [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+ * 8) Alice accepts the friend request.
+ * 9) Alice clicks on Bob in the friends list.
+ * 10) Alice sees Bob's pinned trip.
+ */
+class E2EFriendFlowTest : FirestoreSwissTravelTest() {
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @get:Rule
+  val permissionRule: GrantPermissionRule =
```
[the remark this answers is not in the document]
Comment (AdrienRenggli) [reply app/src/androidTest/java/com/github/swent/swisstravel/e2e/E2EFriendsTest.kt]: It was in case I wanted to do more things when watching the pined trip of someone else.
It would have been needed to allow the location. But I can delete it. Thanks.
